HospitalToday/Services/Implementation/DoctorService.cs

```diff
@@ -27,33 +29,40 @@ public void Remove(Person item)
             personRep.Delete(item.Id);
         }
 
-        public List<Person> GetList()
+        public IList<Person> GetList()
         {
             var persons = personRep.GetList().Where(x => x is Doctor);
-            return persons != null ? persons.ToList() : null;
+            return persons.ToList();
         }
 
-        public Report GetReport(Person doctor, Person patient, List<Medicine> medicines, DateTime date)
+        public Person GetItemById(int id)
+        {
+            return personRep.GetItem(id);
+        }
+
+        public int CreateReport(Person doctor, Person patient, List<Medicine> medicines, DateTime? date)
         {
+            var resultId = -1;
             if (doctor == null || patient == null)
             {
-                return null;
+                return resultId;
```

Comment:
Let's throw an exception and then handle it in outer scope.

Reply:
Added - throw ArgumentExeception